src/webview/webViewEventHandlers.js

```diff
@@ -150,17 +153,28 @@ const handleImage = (props: Props, src: string, messageId: number) => {
   }
 };
 
-const handleLongPress = (props: Props, _: GetText, isHeader: boolean, messageId: number) => {
-  const message = props.messages.find(x => x.id === messageId);
-  if (!message) {
-    return;
+const handleLongPress = (
+  props: Props,
+  _: GetText,
+  target: 'message' | 'header' | 'link',
+  messageId: number,
+  href: string,
+) => {
+  if (href !== null) {
```

Comment:
Cool, but this revision is incomplete -- there's still the redundant distinction of `link` from `message`, appearing in several places in the code.

Reply:
Oh, also the commit message still describes your v1 design:
```
    I have added
    another event to `MessageListEvent` which abstracts a long press on a
    url, and edited the long press listener on `js.js` a little bit.
```

Reply:
I purposely left the distinction between `link` and `message` for the sake of readability of code. Do you think I should remove it?

Fixed the commit message (in WIP)! :)